internal/pkg/term/selector/selector.go

```diff
@@ -176,6 +195,100 @@ func NewDeploySelect(prompt Prompter, configStore ConfigLister, deployStore Depl
 	}
 }
 
+// NewTaskSelect returns a new selector that chooses a running task.
+func NewTaskSelect(prompt Prompter, lister TaskLister) *TaskSelect {
+	return &TaskSelect{
+		prompt: prompt,
+		lister: lister,
+	}
+}
+
+// TaskOpts sets up optional parameters for Task function.
+type TaskOpts func(*TaskSelect)
+
+// WithAppEnv sets up the app name and env name for TaskSelect.
+func WithAppEnv(app, env string) TaskOpts {
+	return func(in *TaskSelect) {
+		in.app = app
+		in.env = env
+	}
+}
+
+// WithDefault uses default cluster for TaskSelect.
+func WithDefault() TaskOpts {
+	return func(in *TaskSelect) {
+		in.defaultCluster = true
+	}
+}
+
+// WithTaskGroup sets up the task group name for TaskSelect.
+func WithTaskGroup(taskGroup string) TaskOpts {
+	return func(in *TaskSelect) {
+		in.taskGroup = taskGroup
+	}
+}
+
+// WithTaskID sets up the task ID for TaskSelect.
+func WithTaskID(id string) TaskOpts {
+	return func(in *TaskSelect) {
+		in.taskID = id
+	}
+}
+
+// RunningTask has the user select a running task. Callers can provide either app and env names,
+// or use default cluster.
+func (s *TaskSelect) RunningTask(prompt, help string, opts ...TaskOpts) (*awsecs.Task, error) {
+	var tasks []*awsecs.Task
+	var err error
+	for _, opt := range opts {
+		opt(s)
+	}
+	filter := ecs.ListTasksFilter{
+		TaskGroup: s.taskGroup,
+		TaskID:    s.taskID,
+	}
+	if s.defaultCluster {
```

Comment:
Hmm im planning to do this validation in task exec `Validate` method. Wondering if it would be duplicated to validate here. And I think it makes sense to show both if both default and app env are specified?

Reply:
Hmm, yeah. I agree, that makes good sense to me.